components/tools/common/ui/RequireWalletConnectionV2.tsx

```diff
@@ -0,0 +1,224 @@
+            const accounts = await window.avalanche.request({ method: 'eth_requestAccounts', params: [] });
+            if (!accounts || accounts.length === 0) {
+                setError('No account detected');
+                return;
+            }
+            const selectedAccount = accounts[0];
+            setAccount(selectedAccount);
+
+            // Check if RPC endpoint is reachable
+            setConnectionStatus('rpc_error');
+            await window.avalanche.request({
+                method: 'eth_blockNumber',
+                params: []
+            });
+
+            setConnectionStatus('correct_chain_connected');
+            onConnection && onConnection();
+        } catch (error) {
+            if (error instanceof Error) {
+                setError(`Connection error: ${error.message}`);
+            } else {
+                setError(`An unknown error occurred: ${error}`);
+            }
+        }
+    };
+
+    useEffect(() => {
+        checkConnection();
+
+        if (window.avalanche) {
+            const provider = window.avalanche!;
+            provider.on('chainChanged', checkConnection);
+            provider.on('accountsChanged', checkConnection);
+
+            return () => {
+                provider.removeListener('chainChanged', checkConnection);
+                provider.removeListener('accountsChanged', checkConnection);
+            };
+        }
+    }, []);
+
+    const switchChain = async () => {
+        if (!window.avalanche) {
+            setConnectionStatus('no_wallet_installed');
+            return;
+        }
+
+        const walletClient = createWalletClient({
+            chain,
+            transport: custom(window.avalanche)
+        });
+
+        try {
+            await walletClient.switchChain({ id: chain.id });
+        } catch (error: any) {
+            if (error.code === 4902) {
+                try {
+                    await walletClient.addChain({ chain });
+                    checkConnection();
+                } catch (addError) {
+                    if (addError instanceof Error) {
+                        setError(`Failed to add network: ${addError.message}`);
+                    } else {
+                        console.error(addError);
+                        setError(`An unknown error occurred: ${addError}`);
+                    }
+                }
+            } else {
+                setError(`Failed to switch network: ${error}`);
+            }
+        }
+    };
+
+    const renderContent = () => {
+        switch (connectionStatus) {
+            case 'no_wallet_installed':
+                return (
+                    <div className="bg-yellow-50 border-l-4 border-yellow-400 p-4 mb-6">
+                        <h3 className="font-medium mb-2">No Wallet Detected</h3>
+                        <p className="mb-4">
+                            You need a Web3 wallet to interact with your L1 chain. We recommend installing Core wallet:
```

Comment:
We should state that core wallet is required:

```suggestion
                            You need an Core Wallet to launch your L1:
```